src/prefect/client/orchestration/artifacts.py

```diff
@@ -0,0 +1,239 @@
+
+    async def delete_artifact(self, artifact_id: "UUID") -> None:
+        try:
+            await self.request(
+                "DELETE",
+                "/artifacts/{artifact_id}",
+                path_params={"artifact_id": artifact_id},
+            )
+        except HTTPStatusError as e:
+            if e.response.status_code == 404:
+                raise ObjectNotFound(http_exc=e) from e
+            else:
+                raise
+
+
+class ArtifactCollectionClient(BaseClient):
+    def read_latest_artifacts(
+        self, **kwargs: Unpack["ArtifactCollectionReadParams"]
+    ) -> list["ArtifactCollection"]:
+        response = self.request(
+            "POST",
+            "/artifacts/latest/filter",
+            json={
+                "artifacts": (
+                    artifact_filter.model_dump(mode="json", exclude_unset=True)
+                    if (artifact_filter := kwargs.get("artifact_filter"))
+                    else None
+                ),
+                "flow_runs": (
+                    flow_run_filter.model_dump(mode="json", exclude_unset=True)
+                    if (flow_run_filter := kwargs.get("flow_run_filter"))
+                    else None
+                ),
+                "task_runs": (
+                    task_run_filter.model_dump(mode="json", exclude_unset=True)
+                    if (task_run_filter := kwargs.get("task_run_filter"))
+                    else None
+                ),
+                "limit": kwargs.get("limit", None),
+                "offset": kwargs.get("offset", 0),
+                "sort": kwargs.get("sort", None),
+            },
+        )
+        from prefect.client.schemas.objects import ArtifactCollection
+
+        return ArtifactCollection.model_validate_list(response.json())
+
+
+class ArtifactCollectionAsyncClient(BaseAsyncClient):
+    async def read_latest_artifacts(
+        self, **kwargs: Unpack["ArtifactCollectionReadParams"]
+    ) -> list["ArtifactCollection"]:
+        response = await self.request(
+            "POST",
+            "/artifacts/latest/filter",
+            json={
+                "artifacts": (
+                    artifact_filter.model_dump(mode="json", exclude_unset=True)
+                    if (artifact_filter := kwargs.get("artifact_filter"))
+                    else None
+                ),
+                "flow_runs": (
+                    flow_run_filter.model_dump(mode="json", exclude_unset=True)
+                    if (flow_run_filter := kwargs.get("flow_run_filter"))
+                    else None
+                ),
+                "task_runs": (
+                    task_run_filter.model_dump(mode="json", exclude_unset=True)
+                    if (task_run_filter := kwargs.get("task_run_filter"))
+                    else None
+                ),
+                "limit": kwargs.get("limit", None),
+                "offset": kwargs.get("offset", 0),
+                "sort": kwargs.get("sort", None),
```

Comment:
If the defaults are declared on the typed dicts at the top of this module, do we still need to provide a default value for the `.get` call here?